src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -408,6 +412,34 @@ protected override void StopProcessing()
 
         #region Helper Methods
 
+        private void ThrowNotSuccessStatusCodeError(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            var detailMsg = "";
+            var reader = new StreamReader(StreamHelper.GetResponseStream(response));
+            try
+            {
+                detailMsg = reader.ReadToEnd();
+            }
+            catch (IOException)
+            {
+            }
+            catch (ArgumentException)
+            {
+            }
+            finally
+            {
+                reader.Dispose();
+            }
+
+            var msg = string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.NotSuccessStatusCode, (int) response.StatusCode, response.ReasonPhrase);
+            ErrorRecord er = new ErrorRecord(new WebException(msg) { Response = response }, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, request);
```

Comment:
@daxian-dbw in both cases we considered we would have trade-offs. However, in the case of ```Exception.Data``` would have a breaking change, that could lead to code structures like this ```$error[0].Exception.Response, $error[0].Data.StatusCode | select -f 1``` ([another "great" example](https://github.com/konstantinvlasenko/PowerSlim/blob/master/slim.ps1#L174-L175)). Moreover, it will be hard to guess without taking a look at the source code or scrutinizing MSDN carefully that the status code was in the Data property, all the previous blog posts, kb articles will be outdated.

From another side, in the case of the Response field, oh yes, we would have that a little bit tricky exception logic in the CoreClr implementation, but all that logic would be encapsulated in that method and with tests we could get along with that.

Does anybody have third solution for this problem:))?